.github/workflows/build.yml

```diff
@@ -0,0 +1,97 @@
+name: Build, push and deploy faucet image
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - alephzero
+    paths-ignore:
+      - ".github/**"
+      - "*.md"
+
+jobs:
+  deploy-faucet:
+    name: Build and push faucet image
+    runs-on: ubuntu-latest
+    steps:
+      - name: GIT | Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.9.0
+        with:
+          access_token: ${{ github.token }}
```

Comment:
If you want to prevent running multiple workflows for the same branches I think it would be better to use concurrency.
Examle:
```
concurrency:
  group: ${{ github.ref }}-${{ github.workflow }}
  cancel-in-progress: true
```

Reply:
Good point